middle_end/flambda2/tests/tools/flexpect.ml

```diff
@@ -41,6 +41,7 @@ let run_expect_test ~get_module_info ~extension ~filename
   let cmx_loader = Flambda_cmx.create_loader ~get_module_info in
   let { Simplify.unit = actual_fl; _ } =
     Simplify.run ~cmx_loader ~round:0 before_fl
+      ~code_slot_offsets:Code_id.Map.empty
```

Comment:
I'd much rather pass the correct slot offsets to the simplifier here. Then we could add #1646 as a test case.

Reply:
I suspect that we can't actually, because examples like the one in #1646 require correct free names computations in addition to correct offsets.
I think it would be easier to fix `fexpr_to_flambda` to correctly compute free names, then reuse the mechanism for computing slot offsets too.

Reply:
As @lthls said, if we want correct slot offsets here, then we'd need to compute them properly in a dedicated pass (probably along with free_names), and that would probably be better in a separate PR (in the sense this this PR does not make things worse than before for flexpect, but fixes issue in Simplify, and another PR could upgrade flexpect to the same level as Simplify).

Reply:
Is there anything stopping us from adding the slot offsets to fexpr syntax and passing in the correct ones here? Whether or not we add them to `Flambda_unit.t` directly, we could have `Flambda_to_fexpr` store them and have `Fexpr_to_flambda` return `Flambda_unit.t * Slot_offsets.t Code_id.Map.t`

Reply:
I guess you could do that ? It seems more complicated to me than just computing them when rebuilding the flambda term, but if you think it's simpler this way I'm fine with that.
It still doesn't allow us to handle the example from #1646 though, because of the missing free names.

Reply:
I don't think it's a good idea to add the slot offsets to the syntax: it'd be a bit like adding the free_names to the syntax: it can and should be computed from the term, if only to avoid false/absurd/incorrect slot offsets in the input of tests.

Reply:
In any case:
- flexpect was already broken on that example, this PR only fixes Simplify
- this PR is already not trivial, so I'd rather not complexify it by adding a lot of code for flexpect
- I think it's better to fix flexpect in a separate PR that would add the proper traversal to compute the free_names and slot_offsets for flexpect. 
@lukemaurer what do you think ?

Reply:
Since this PR fixes a bug, and after discussion with @mshinwell , I left a CR here to remind us to that sometime.